rs/xnet/payload_builder/src/lib.rs

```diff
@@ -879,11 +883,24 @@ impl XNetPayloadBuilderImpl {
 ///
 /// In order to prevent mutual stalling, only applies to incoming NNS
 /// streams; and to `Application`-subnet-to-`System`-subnet streams.
-pub fn get_msg_limit(subnet_id: SubnetId, state: &ReplicatedState) -> Option<usize> {
+pub fn get_msg_limit(
+    subnet_id: SubnetId,
+    state: &ReplicatedState,
+    expected: Option<&ExpectedIndices>,
+    slice_begin: StreamIndex,
```

Comment:
Nit: This is `stream_begin`, not `slice_begin`.

Reply:
This comment I don't understand. I called it `slice_begin` because that is what is used for [garbage collecting signals](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/messaging/src/routing/stream_handler.rs?L528).

Reply:
As you observed on Slack, `stream_slice.header().begin()` is the beginning of the stream, not of the slice. The header in question is the stream header. A slice consists of the stream header and some range of messages. The stream header would be the same regardless of which messages are included in the slice (or if no messages are included).

I.e. this is the stream begin, as certified by the remote subnet. Not the slice begin.